src/xo-app/pool/tab-logs.js

```diff
@@ -1,111 +1,57 @@
 import _ from 'intl'
-import ActionRow from 'action-row-button'
 import React, { Component } from 'react'
-import TabButton from 'tab-button'
+import SortedTable from 'sorted-table'
 import { deleteMessage } from 'xo'
-import { createPager, createSelector } from 'selectors'
 import { FormattedRelative, FormattedTime } from 'react-intl'
-import { Container, Row, Col } from 'grid'
 import {
-  ceil,
-  isEmpty,
   map
 } from 'lodash'
 
-const LOGS_PER_PAGE = 10
-
-export default class TabLogs extends Component {
-  constructor () {
-    super()
-
-    this.getLogs = createPager(
-      () => this.props.logs,
-      () => this.state.page,
-      LOGS_PER_PAGE
-    )
-
-    this.getNPages = createSelector(
-      () => this.props.logs ? this.props.logs.length : 0,
-      nLogs => ceil(nLogs / LOGS_PER_PAGE)
-    )
+const LOG_COLUMNS = [
+  {
+    default: true,
+    itemRenderer: log => <div>
+      <FormattedTime value={log.time * 1000} minute='numeric' hour='numeric' day='numeric' month='long' year='numeric' /> (<FormattedRelative value={log.time * 1000} />)
+    </div>,
+    name: _('logDate'),
+    sortCriteria: 'time'
+  },
+  {
+    itemRenderer: log => log.name,
+    name: _('logName'),
+    sortCriteria: 'name'
+  },
+  {
+    itemRenderer: log => log.body,
+    name: _('logContent'),
+    sortCriteria: 'body'
+  }
+]
 
-    this.state = {
-      page: 1
-    }
+const INDIVIDUAL_ACTIONS = [
+  {
+    handler: deleteMessage,
+    icon: 'delete',
+    label: _('logDelete')
   }
+]
 
-  _deleteAllLogs = () => map(this.props.logs, deleteMessage)
-  _nextPage = () => this.setState({ page: Math.min(this.state.page + 1, this.getNPages()) })
-  _previousPage = () => this.setState({ page: Math.max(this.state.page - 1, 1) })
+const GROUPED_ACTIONS = [
+  {
+    handler: logs => Promise.all(map(logs, deleteMessage)),
```

Comment:
Maybe you could add a `confirm` modal before deleting multiple logs at once. Same for the VM logs.

@olivierlambert What do you think?

Reply:
Yes, a confirmation modal is a good idea here